### PR TITLE
Fix resumed live-game substitution dropdown state

### LINKED ISSUE #526: Add resumed-live-game substitution coverage for Game Day lineup controls
## Test Coverage Gap

- Workflow Focus: Test Game Day
- Type: targeted test gap

### Workflow
Game Day Command Center, reopen a live game after substitutions were already recorded

### Risk Summary
A coach can reopen a live game, see the field diagram correctly reflect prior substitutions, but get incorrect OUT and IN dropdown choices based on the original plan. That can lead to logging a substitution against the wrong player or masking state drift until a real game is in progress.

### Coverage Gap
There is no automated test that loads game-day state with both saved gamePlan and saved rotationActual, then verifies the substitution controls use the merged on-field state. Current coverage only checks lineup reconstruction and on-field rendering helpers, not the resumed live substitution workflow.

### Suggested Tests
1. Load game-day with gamePlan lineups showing Bob on the field and rotationActual showing Carol already subbed in for Bob, then verify the field diagram and the substitution OUT dropdown both show Carol as on-field and Bob as off-field after initial render.
2. Simulate reopening a live game mid-period with prior substitutions persisted, then verify the IN dropdown excludes players already subbed onto the field and the next substitution logs against the current position occupant, not the stale planned player.

### Code Evidence
- `game-day.html:1007 loads state.rotationPlan from game.gamePlan and separately hydrates state.rotationActual from saved game data, creating a resume path that depends on both sources staying aligned`
- `game-day.html:1980 buildOnFieldMap merges rotationActual over rotationPlan for the field diagram, so the visible lineup can differ from the original plan after reload`
- `game-day.html:2099 populateSubDropdowns builds on-field and off-field lists from state.rotationPlan only, ignoring rotationActual entirely`
- `game-day.html:2127 applySub mutates state.rotationPlan locally after a new sub, which makes same-session happy-path testing pass and hides reload-only regressions`
- `test-game-day.html:387 only exercises buildOnFieldMap helper behavior, and does not test populateSubDropdowns or resumed live-game substitution state`

### Why Now
This is a high-value gap because it targets a real coach workflow under live-game pressure, and the current code has split sources of truth that are easy to miss in manual testing. One focused test here is likely to catch a user-facing regression before the next Game Day change ships.

---
Detected by `paulbot-bughunter` fallback after duplicate-safe bug scan.

### PR DESCRIPTION
**Pull request by pauljsnider**
Closes #526

## What changed
- updated Game Day substitution controls to derive OUT and IN choices from the merged on-field state, not just the original saved rotation plan
- updated follow-up substitution handling to resolve the outgoing player's position from the current merged lineup after a live game is reopened
- added regression coverage for resumed live-game substitution dropdowns and position targeting

## Why
When a live game was reopened after prior substitutions, the field diagram reflected `rotationActual` but the substitution controls still read from the stale planned lineup. That could show the wrong OUT and IN choices and log the next substitution against the wrong player position.

### CODE REVIEW

**Comment by pauljsnider** [comment]
### Mode
reply

### Decision
No code change required.

### Evidence
- `gh pr view 532 --repo pauljsnider/allplays --json number,title,headRefName,reviews,commits,mergeStateStatus,url`
  - PR head branch: `paulbot/fix/issue-526-202604131914`
  - Reviewed commit: `ecd01dfa9a0b56ee55f4279a945b99adc69631eb`
  - Amazon Q review state: `COMMENTED`
  - Review summary: “No blocking defects found.”
- `git status --short --branch`
  - Local workspace is not on the PR head branch and contains unrelated dirty state, so no branch-local verification or edits were performed from this checkout.

### Code Changes
None.

### Role Summaries
- Requirements: no additional requirement raised by the review.
- Architecture: no design concern identified.
- QA: no defect requiring test-driven follow-up.
- Code: no edit, commit, or push required.

### Validation
Commands run:
```bash
gh pr view 532 --repo pauljsnider/allplays --json number,title,headRefName,reviews,commits,mergeStateStatus,url
gh pr checks 532 --repo pauljsnider/allplays
git status --short --branch
```

Impacted workflows verified:
- `Amazon Q Developer` ✅
- `cache-bust-guard` ✅
- `deploy` ✅
- `preview-smoke` ✅
- `unit-tests` ✅

No local test commands were run because no code changes were required.

File: game-day.html [hunk 1]
```diff
@@ -2120,8 +2120,8 @@
             const outSel = document.getElementById('sub-out-select');
             const inSel = document.getElementById('sub-in-select');
             const period = state.activePeriodGD;
-            const planForPeriod = state.rotationPlan?.[period] || {};
-            const onFieldIds = new Set(Object.values(planForPeriod).filter(Boolean));
+            const onField = buildOnFieldMap(period);
+            const onFieldIds = new Set(Object.values(onField).filter(Boolean));
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply game-day.html]: ### Remediation Update

Addressed this review feedback pass on PR #532.

- Commit: `72e2d68fe74a6c4ecaae21e13bfdb291d6253159`
- Validation: not-run

{"success":true,"files_changed":["docs/pr-notes/runs/532-remediator-20260413T192509Z/architecture.md","docs/pr-notes/runs/532-remediator-20260413T192509Z/code-plan.md","docs/pr-notes/runs/532-remediator-20260413T192509Z/qa.md","docs/pr-notes/runs/532-remediator-20260413T192509Z/requirements.md","game-day.html","test-game-day.html"],"commit_message":"Keep substitution state keyed by player ID","threads_addressed":["PRRT_kwDOQe-T5856oIZJ"],"summary":"Updated game-day substitution replay to persist and resolve players by stable player IDs, while keeping a legacy name-based fallback for older saved games. This keeps the on-field map and substitution dropdowns accurate after reloads, even when players share a display name. I also mirrored the logic in the game-day test page and added coverage for duplicate-name and backward-compatibility cases."}

**Comment by pauljsnider** [comment]
### CI Fix Update

Addressed failing CI checks on PR #532.

- Commit: `607d6988ef853e3afb4b43f67f264e8ecaf59a32`
- Fixed checks: unit-tests, unit-tests
- Validation: not-run

{"success":true,"files_changed":["tests/unit/game-day-live-substitutions.test.js","docs/pr-notes/runs/532-ci-fix-20260418T214507Z/architecture.md","docs/pr-notes/runs/532-ci-fix-20260418T214507Z/qa.md","docs/pr-notes/runs/532-ci-fix-20260418T214507Z/code-plan.md"],"commit_message":"fix:address-ci-failure: update live substitution test expectations","checks_addressed":["unit-tests [deploy-preview]","unit-tests [ci]"],"summary":"The CI failure was caused by stale unit tests, not broken runtime behavior. `js/game-day-live-substitutions.js` now persists stable player ID fields on substitution records, but `tests/unit/game-day-live-substitutions.test.js` still deep-equaled the old name-only payload shape. I updated the two failing assertions to validate the persisted ID fields and confirmed the affected Vitest file passes locally."}
